### ISSUE TITLE
plot_spectrum_datasets_off_regions. props.pop('color') but color might not be in props

### ISSUE THREAD

**Issue by mireianievas** [gammapy/gammapy#4489]
**Gammapy version**
1.0.1

**Bug description**

In visualization/utils.py I see

```
L90: plot_kwargs.setdefault("edgecolor", props.pop("color"))
```

however, color might not be in props, raising a KeyError. Found while running plot_spectrum_datasets_off_regions. 

**Expected behavior**
before popping 'color' we should make sure props include it. 

**To Reproduce**
Long run dataset, running https://docs.gammapy.org/1.0.1/user-guide/makers/reflected.html?highlight=reflected%20regions%20background

**Comment by registerrier** [comment]
Thanks @mireianievas . I think this is a duplicate from #4222  which has been solved in #4397

**Comment by mireianievas** [comment]
Thanks, and sorry for having missed it,.
